### PR TITLE
Implement link embeds for articles using Cloudflare worker

### PR DESCRIPTION
**Pull request by lukebrody**
image

- Worker rewrites OpenGraph headers for articles, comparisons, stats.
- Worker implements a route to render dynamic images for articles. (See diff for example)
  -  Does vector basemap rendering in the worker to match site style and exclude labels which are cluttered in this context
- Implement dev panel to test out embeds
- Implement automatic deploy of worker from densitydb

## Migration steps

- Mirror existing DNS records to Cloudflare
  > Add the zone in Cloudflare and import every existing record before switching nameservers at the registrar — the Mailgun MX pair and the SPF TXT record included, or mail breaks along with the cutover. Then set SSL/TLS to Full, proxy the four GitHub Pages A records, and leave Rocket Loader and HTML minification off so nothing rewrites the pages underneath the Worker.
- Cut over nameserver to Cloudflare
- Add deploy secrets to densitydb

### CODE REVIEW

File: react/og-worker/src/map-layout.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    /** The box's top-left corner, in that same scaled space. */
+    originX: number
+    originY: number
+}
+
+export function fitRings(rings: Ring[], width: number, height: number): MapLayout {
+    const projected = rings.flat().map(project)
+    const xs = projected.map(p => p[0])
+    const ys = projected.map(p => p[1])
+    const [minX, maxX] = [Math.min(...xs), Math.max(...xs)]
```
[the remark this answers is not in the document]
Comment (lukebrody) [reply react/og-worker/src/map-layout.ts]: This didn't reproduce, but added an edge case test
